### PR TITLE
feat: Add slack connector

### PR DESCRIPTION
**Pull request by tauffredou**

### CODE REVIEW

File: bin/slack/main.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+#             ]
+#         }
+#     ]
+# }
+
+conn = CI::Connector.from_env
+conn.on('environment.lifecycle') do |event|
+  if event['event'] == 'release_completed'
+
+    uri = URI("https://hooks.slack.com/services/T03AA3Y27/BFP2H3YV9/" + ENV['SLACK_TOKEN'])
```
Comment (gearnode) [inline bin/slack/main.rb]: What's about using environment to configure this?

File: bin/slack/main.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    header = {
+      'Content-Type': 'application/json',
+    }
+
+    payload = {
+      text: "Release #{event['data']['project']['name']} #{event['data']['name']}",
```
Comment (gearnode) [inline bin/slack/main.rb]: With this method we can have raise condition when the event does not contain the deep key. Ruby has introduce safe operator to dig in deep key on a hash object. This method was name `dig`, ref: https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig

File: bin/slack/main.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+              "short": false
+            }
+          ]
+        }
+      ]
+    }
+
+    # Create the HTTP objects
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
```
Comment (gearnode) [inline bin/slack/main.rb]: 👍

File: bin/slack/main.rb [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    request = Net::HTTP::Post.new(uri.request_uri, header)
+    puts payload.to_json
+    request.body = payload.to_json
+
+    # Send the request
+    http.request(request)
```
Comment (gearnode) [inline bin/slack/main.rb]: Why we don't check the response from slack here?

Comment (tauffredou) [reply bin/slack/main.rb]: no, I don't have kafka ack either

File: bin/slack/main.rb [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+#                     "value": "bad"
+#                 }
+#             ]
+#         }
+#     ]
+# }
+
+conn = CI::Connector.from_env
+conn.on('environment.lifecycle') do |event|
+  if event['event'] == 'release_completed'
```
Comment (gearnode) [inline bin/slack/main.rb]: You can use early return here e.g.

```
continue if event['event'] != 'release_completed'
```

File: bin/slack/main.rb (hunk 4 above)
Comment (gearnode) [reply bin/slack/main.rb]: @tauffredou How we get error notification when this post on slack failed?

File: bin/slack/main.rb (hunk 2 above)
Comment (tauffredou) [reply bin/slack/main.rb]: Type is loosly checked before, so I have those fields
Event with dig, what to do with the error ?

Tks for the pointer

File: bin/slack/main.rb (hunk 1 above)
Comment (tauffredou) [reply bin/slack/main.rb]: :+1:

**Pull request merged**